hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -55,16 +69,82 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         return Response.newBuilder().transactionGetReceipt(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var recordCache = context.createStore(RecordCache.class);
+        final TransactionGetReceiptQuery transactionGetReceiptQuery = query.transactionGetReceiptOrThrow();
```

Comment:
Also, this is totally not a necessary thing to change, but personally I'd pick a shorter variable name for `transactionGetReceiptQuery`. I would probably have done `final var query = context.query().transactionGetReceiptOrThrow()` as a single line, since the `query` variable is not used again. Or maybe instead of calling it "query" maybe I would use `queryBody` or `op` to align with the concepts used on the transaction handling side.

Reply:
The style might be something to discuss with platform, which took the opposite approach and prefer not to use var.

Reply:
I feel the code is more readable that way